bats/foreign-keys.bats

```diff
@@ -1894,3 +1894,282 @@ SQL
     dolt add super_parent
     dolt commit -m "passes now"
 }
+
+@test "foreign-keys: Merge valid onto parent" {
+    dolt sql <<SQL
+CREATE TABLE parent (
+  pk BIGINT PRIMARY KEY,
+  v1 BIGINT
+);
+CREATE TABLE child (
+  pk BIGINT PRIMARY KEY,
+  v1 BIGINT,
+  CONSTRAINT fk_name FOREIGN KEY (v1)
+    REFERENCES parent(v1)
+);
+INSERT INTO parent VALUES (1, 1), (2, 2), (3, 3);
+INSERT INTO child VALUES (2, 1), (3, 2), (4, 3);
+SQL
+
+    dolt add -A
+    dolt commit -m "initial commit"
+    dolt checkout -b other
+    dolt checkout master
+    dolt sql <<SQL
+INSERT INTO parent VALUES (4, 3);
+SQL
+    dolt add -A
+    dolt commit -m "added row"
+    dolt checkout other
+
+    dolt sql <<SQL
+ALTER TABLE child DROP FOREIGN KEY fk_name;
+UPDATE parent SET v1 = v1 - 1;
+SQL
+    dolt add -A
+    dolt commit -m "updated parent"
+    dolt checkout master
+    dolt merge other
+    
+    run dolt sql -q "SELECT * FROM parent" -r=csv
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "pk,v1" ]] || false
+    [[ "$output" =~ "1,0" ]] || false
+    [[ "$output" =~ "2,1" ]] || false
+    [[ "$output" =~ "3,2" ]] || false
+    [[ "$output" =~ "4,3" ]] || false
+    [[ "${#lines[@]}" = "5" ]] || false
+    run dolt sql -q "SELECT * FROM child" -r=csv
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "pk,v1" ]] || false
+    [[ "$output" =~ "2,1" ]] || false
+    [[ "$output" =~ "3,2" ]] || false
+    [[ "$output" =~ "4,3" ]] || false
+    [[ "${#lines[@]}" = "4" ]] || false
+}
+
+@test "foreign-keys: Merge invalid onto parent" {
+    dolt sql <<SQL
+CREATE TABLE parent (
+  pk BIGINT PRIMARY KEY,
+  v1 BIGINT
+);
+CREATE TABLE child (
+  pk BIGINT PRIMARY KEY,
+  v1 BIGINT,
+  CONSTRAINT fk_name FOREIGN KEY (v1)
+    REFERENCES parent(v1)
+);
+INSERT INTO parent VALUES (1, 1), (2, 2), (3, 3);
+INSERT INTO child VALUES (2, 1), (3, 2), (4, 3);
+SQL
+
+    dolt add -A
+    dolt commit -m "initial commit"
+    dolt checkout -b other
+    dolt checkout master
+    dolt sql <<SQL
+INSERT INTO parent VALUES (4, 4);
+SQL
+    dolt add -A
+    dolt commit -m "added row"
+    dolt checkout other
+
+    dolt sql <<SQL
+ALTER TABLE child DROP FOREIGN KEY fk_name;

```

Comment:
This is actually kind of weird... I would expect the schema change (dropped a foreign key) to get merged along with the data, so that there wouldn't actually be a merge validation problem. I would add a note about this behavior -- it seems wrong to me.

Reply:
In the meantime, you could use a flag or session var instead to disable foreign key checks, right?